tests/parser.cpp

```diff
@@ -26,35 +26,40 @@ void test_parse_failure(BPFtrace &bpftrace,
   }
 }
 
-void test_parse_failure(const std::string &input,
+void test_parse_failure(std::string_view input,
                         std::string_view expected_error = {})
 {
   BPFtrace bpftrace;
   test_parse_failure(bpftrace, input, expected_error);
 }
 
-void test(BPFtrace &bpftrace,
-          const std::string &input,
-          const std::string &output)
+void test(BPFtrace &bpftrace, std::string_view input, std::string_view expected)
 {
   Driver driver(bpftrace);
-  ASSERT_EQ(driver.parse_str(input), 0);
+  EXPECT_EQ(driver.parse_str(input), 0);
+
+  if (expected[0] == '\n')
+    expected.remove_prefix(1); // Remove initial '\n'
 
   std::ostringstream out;
   Printer printer(out);
   printer.print(driver.root.get());
-  EXPECT_EQ(output, out.str());
+  EXPECT_EQ(expected, out.str());
 }
 
-void test(const std::string &input, const std::string &output)
+void test(std::string_view input, std::string_view expected)
 {
   BPFtrace bpftrace;
-  test(bpftrace, input, output);
+  test(bpftrace, input, expected);
 }
 
 TEST(Parser, builtin_variables)
 {
-  test("kprobe:f { pid }", "Program\n kprobe:f\n  builtin: pid\n");
+  test("kprobe:f { pid }", R"(
+Program
+ kprobe:f
+  builtin: pid
+)");
```

Comment:
Any reason not to convert the below test cases, too?

Reply:
Laziness! I just converted the first one to give an example of the new functionality. What I really wanted this feature for is the new tests I'm writing for #2955